services/svc_simpleurl.py

```diff
@@ -23,25 +27,65 @@ class SimpleUrl(BaseService):
   def __init__(self, configDir, id, name):
     BaseService.__init__(self, configDir, id, name, needConfig=False, needOAuth=False)
 
-
   def helpKeywords(self):
     return 'Each item is a URL that should return a single image. The URL may contain the terms "{width}" and/or "{height}" which will be replaced by numbers describing the size of the display.'
 
-
-  def prepareNextItem(self, destinationFile, supportedMimeTypes, displaySize):
+  def prepareNextItem(self, destinationDir, supportedMimeTypes, displaySize, randomize):
```

Comment:
This one would certainly benefit from the abstraction mentioned above since it grew quite a bit to support the new features.

Reply:
yes it does. 
I kept it to distinguish between services that `needKeywords()` (-->`selectImageFromAlbum`) and services that don't (no such service implemented yet).